### PR TITLE
fix: add checking for Blake256 parameters

### PR DESCRIPTION
**Pull request by DenisKolodin**
It adds checking for `Blake256` parameters: salt and persona should be less or equal 16 bytes length.

Motivation: the fixes for #110.

### CODE REVIEW

File: src/hash/blake2.rs [hunk 1]
```diff
@@ -12,19 +12,25 @@ use digest::{
...
-    pub fn with_params(key: &[u8], salt: &[u8], persona: &[u8]) -> Self {
-        Self(VarBlake2b::with_params(
-            key,
-            salt,
-            persona,
-            <Self as FixedOutput>::OutputSize::USIZE,
-        ))
+    pub fn with_params(key: &[u8], salt: &[u8], persona: &[u8]) -> Result<Self, HashError> {
+        if salt.len() > 16 || persona.len() > 16 {
+            Err(HashError::WorngLength)
```
Comment (hansieodendaal) [inline src/hash/blake2.rs:25]:
```suggestion
            Err(HashError::WrongLength)
```

**Review by hansieodendaal — DISMISSED** [review]
LGTM

**Comment by hansieodendaal** [comment]
I prop0se we add another function to mae full use of the flexibility provided by `VarBlake2b::with_params`

``` rust
pub fn with_params_var_size(key: &[u8], salt: &[u8], persona: &[u8], output_size: usize) -> Result<Self, HashError>
```

File: src/hash/blake2.rs [hunk 2]
```diff
@@ -5,26 +5,37 @@
...
-            salt,
-            persona,
-            <Self as FixedOutput>::OutputSize::USIZE,
-        ))
+    pub fn with_params(key: &[u8], salt: &[u8], persona: &[u8]) -> Result<Self, HashError> {
+        Self::with_params_var_size(key, salt, persona, <Self as FixedOutput>::OutputSize::USIZE)
+    }
+
+    /// Constructs a `Blake256` hashing context with an explicitly specified output size.
+    pub fn with_params_var_size(
```
Comment (hansieodendaal) [inline src/hash/blake2.rs:28]: I think we need a unit test for this, with all the edge cases.

**Review by stringhandler — APPROVED** [review]
